File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
   vector<int> ans;
   int answer = 0;
   int mv =0;
   int j=0;
   int k = 0;

   for (int i = 0; i < moves.size(); ++i) {
      mv = moves[i] - 1;
      j = 0;
      while (j < board.size()) {
         if (board[j][mv] == 0) j++;
         else {
            ans.push_back(board[j][mv]);
            board[j][mv] = 0;
            break;
         }
      }
      
      if (ans.size() > 1) {
         while (k+1 < ans.size()) {

            if (ans[k] == ans[k + 1]) {
               ans.erase(ans.begin() + k);
               ans.erase(ans.begin() + k);
               k = 0;
               answer += 2;
            }
            else ++k;
         }
      }
   }

   return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
bucket = []
answer = 0

def _pick(board, number):
    item = 0
    for idx in range(len(board[number])):
        picked_item = board[number][idx]
        if picked_item != 0:
            item = picked_item
            board[number][idx] = 0
            return item
    return item

def pick(board, number):
    global answer
    item = _pick(board, number)
    
    if item == 0:
        return
    
    if len(bucket) and bucket[-1] == item:
        bucket.pop()
        answer+=2
        return
    
    bucket.append(item)
    return

def solution(board, moves):
    board = [list(x) for x in zip(*board)]
    
    for move in moves:
        pick(board, move - 1)
    return answer